src/BSON/Symbol.c

```diff
@@ -0,0 +1,375 @@
+#if PHP_VERSION_ID >= 70000
+	zval                 retval;
+#else
+	zval                *retval;
+#endif
+	php_phongo_bson_state state = PHONGO_BSON_STATE_INITIALIZER;
+
+	php_serialize_data_t      var_hash;
+	smart_str                 buf = { 0 };
+
+	intern = Z_SYMBOL_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+#if PHP_VERSION_ID >= 70000
+	array_init_size(&retval, 1);
+	ADD_ASSOC_STRINGL(&retval, "symbol", intern->symbol, intern->symbol_len);
+#else
+	ALLOC_INIT_ZVAL(retval);
+	array_init_size(retval, 1);
+	ADD_ASSOC_STRINGL(retval, "symbol", intern->symbol, intern->symbol_len);
+#endif
+
+	PHP_VAR_SERIALIZE_INIT(var_hash);
+	php_var_serialize(&buf, &retval, &var_hash TSRMLS_CC);
+	smart_str_0(&buf);
+	PHP_VAR_SERIALIZE_DESTROY(var_hash);
+
+	PHONGO_RETVAL_SMART_STR(buf);
+
+	smart_str_free(&buf);
+	zval_ptr_dtor(&retval);
+	zval_ptr_dtor(&state.zchild);
+} /* }}} */
+
+/* {{{ proto void MongoDB\BSON\Symbol::unserialize(string $serialized)
+*/
+static PHP_METHOD(Symbol, unserialize)
+{
+	php_phongo_symbol_t    *intern;
+	zend_error_handling     error_handling;
+	char                   *serialized;
+	phongo_zpp_char_len     serialized_len;
+#if PHP_VERSION_ID >= 70000
+	zval                    props;
+#else
+	zval                   *props;
+#endif
+	php_unserialize_data_t  var_hash;
+
+	intern = Z_SYMBOL_OBJ_P(getThis());
+
+	zend_replace_error_handling(EH_THROW, phongo_exception_from_phongo_domain(PHONGO_ERROR_INVALID_ARGUMENT), &error_handling TSRMLS_CC);
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s", &serialized, &serialized_len) == FAILURE) {
+		zend_restore_error_handling(&error_handling TSRMLS_CC);
+		return;
+	}
+	zend_restore_error_handling(&error_handling TSRMLS_CC);
+
+#if PHP_VERSION_ID < 70000
+	ALLOC_INIT_ZVAL(props);
+#endif
+	PHP_VAR_UNSERIALIZE_INIT(var_hash);
+	if (!php_var_unserialize(&props, (const unsigned char**) &serialized, (unsigned char *) serialized + serialized_len, &var_hash TSRMLS_CC)) {
+		zval_ptr_dtor(&props);
+		phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "%s unserialization failed", ZSTR_VAL(php_phongo_symbol_ce->name));
+
+		PHP_VAR_UNSERIALIZE_DESTROY(var_hash);
+		return;
+	}
+	PHP_VAR_UNSERIALIZE_DESTROY(var_hash);
+
+#if PHP_VERSION_ID >= 70000
+	php_phongo_symbol_init_from_hash(intern, HASH_OF(&props) TSRMLS_CC);
+#else
+	php_phongo_symbol_init_from_hash(intern, HASH_OF(props) TSRMLS_CC);
+#endif
+	zval_ptr_dtor(&props);
+} /* }}} */
+
+/* {{{ MongoDB\BSON\Symbol function entries */
+ZEND_BEGIN_ARG_INFO_EX(ai_Symbol___set_state, 0, 0, 1)
+	ZEND_ARG_ARRAY_INFO(0, properties, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(ai_Symbol_unserialize, 0, 0, 1)
+	ZEND_ARG_INFO(0, serialized)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(ai_Symbol_void, 0, 0, 0)
+ZEND_END_ARG_INFO()
+
+static zend_function_entry php_phongo_symbol_me[] = {
+	PHP_ME(Symbol, __set_state, ai_Symbol___set_state, ZEND_ACC_PUBLIC|ZEND_ACC_STATIC)
```

Comment:
Does it make sense to implement this method instead of disabling it as we do the constructor?

With this gone, that would leave `unserialize()` as the only way for users to manually create these deprecated objects.  I'm not suggesting we remove `unserialize()`, as folks should be able to round-trip these classes through PHP serialization :)

Reply:
Removed the __set_state implementation all together.